1-js/03-code-quality/04-ninja-code/article.md

```diff
@@ -1,240 +1,238 @@
 
-## Use abbreviations
+## Sử dụng chữ viết tắt
 
-If the team rules forbid the use of one-letter and vague names -- shorten them, make abbreviations.
+Nếu quy tắc của nhóm cấm sử dụng tên có một chữ cái và tên mơ hồ -- hãy rút ngắn chúng, hãy viết tắt.
 
-Like this:
+Như thế này:
 
 - `list` -> `lst`.
 - `userAgent` -> `ua`.
 - `browser` -> `brsr`.
 - ...etc
 
-Only the one with truly good intuition will be able to understand such names. Try to shorten everything. Only a worthy person should be able to uphold the development of your code.
+Chỉ những người có trực giác thực sự tốt mới có thể hiểu được những cái tên như vậy. Cố gắng rút ngắn mọi thứ. Chỉ một người xứng đáng mới có thể duy trì sự phát triển mã của bạn.
 
-## Soar high. Be abstract.
+## Bay cao. Hãy trừu tượng.
 
-```quote author="Laozi (Tao Te Ching)"
-The great square is cornerless<br>
-The great vessel is last complete,<br>
-The great note is rarified sound,<br>
-The great image has no form.
+```quote author="Lão Tử (Đạo đức kinh)"
+Quảng trường lớn không có góc<br>
+Con tàu vĩ đại đã hoàn thành lần cuối,<br>
+Nốt tuyệt vời là âm thanh quý hiếm,<br>
+Hình ảnh tuyệt vời không có hình thức.
 ```
 
-While choosing a name try to use the most abstract word. Like `obj`, `data`, `value`, `item`, `elem` and so on.
+Trong khi chọn tên, hãy cố gắng sử dụng từ trừu tượng nhất. Như `obj`, `data`, `value`, `item`, `elem`, v.v.
 
-- **The ideal name for a variable is `data`.** Use it everywhere you can. Indeed, every variable holds *data*, right?
+- **Tên lý tưởng cho một biến là `data`.** Sử dụng nó ở mọi nơi bạn có thể. Thật vậy, mọi biến đều chứa *dữ liệu*, phải không?
 
-    ...But what to do if `data` is already taken? Try `value`, it's also universal. After all, a variable eventually gets a *value*.
+     ...Nhưng phải làm gì nếu `data` đã được sử dụng? Hãy thử `value`, nó cũng phổ biến. Xét cho cùng, một biến cuối cùng sẽ nhận được *giá trị*.
 
-- **Name a variable by its type: `str`, `num`...**
+- **Đặt tên biến theo kiểu: `str`, `num`...**
 
-    Give them a try. A young initiate may wonder -- are such names really useful for a ninja? Indeed, they are!
+     Cung cấp cho họ một thử. Một người mới bắt đầu có thể thắc mắc - những cái tên như vậy có thực sự hữu ích cho một ninja không? Thật sự, chúng có!
 
-    Sure, the variable name still means something. It says what's inside the variable: a string, a number or something else. But when an outsider tries to understand the code, they'll be surprised to see that there's actually no information at all! And will ultimately fail to alter your well-thought code.
+     Chắc chắn, tên biến vẫn có ý nghĩa gì đó. Nó cho biết những gì bên trong biến: một chuỗi, một số hoặc thứ gì đó khác. Nhưng khi một người ngoài cuộc cố gắng hiểu mã, họ sẽ ngạc nhiên khi thấy rằng thực tế không có thông tin gì cả! Và cuối cùng sẽ không thể thay đổi mã được suy nghĩ kỹ lưỡng của bạn.
```

Comment:
"And will ultimately fail to alter your well-thought code" bạn dịch khó hiểu quá
Nên dịch là "Và cuối cùng họ sẽ thất bại trong việc thay đổi đoạn code mà bạn đã suy nghĩ kỹ lưỡng để viết ra"